doc/glossary.rst

```diff
@@ -1367,6 +1367,13 @@ Methods
         If the estimator was not already :term:`fitted`, calling this method
         should raise a :class:`exceptions.NotFittedError`.
 
+        During transformation, the order of the input samples may not be
```

Comment:
The ordering *must* be preserved

Reply:
Or are you talking about memory layout order?

Reply:
The intent is to address memory layout order. Is the current definition incorrect? Please let me know if "During transformation, the memory layout order of the the input samples may not be preserved," will satisfy your concerns for this PR. 

I am impartial to the verbiage, just want to get it resolved. @jnothman

Reply:
That's okay, but don't call it "the order of the input samples." How about "Unless an in-place operation can be performed (with copy=False), aspects of the input format -- such as number of features, memory layout order, and dtype -- may or may not be preserved by transform."

This is quite a weak statement! It might be better if we analysed what transformers actually do preserve things and documented it better..